sparkplug_plugin/sparkplug_b_input.go

```
@@ -0,0 +1,1135 @@
+		}
+
+		msg := s.createMessageFromMetric(metric, payload, msgType, deviceKey, topicInfo, originalTopic)
+		if msg != nil {
+			batch = append(batch, msg)
+		}
+	}
+
+	return batch
+}
+
+func (s *sparkplugInput) createSingleMessage(payload *sproto.Payload, msgType, deviceKey string, topicInfo *TopicInfo, originalTopic string) service.MessageBatch {
```

golangci-lint (1.64.8)

func `(*sparkplugInput).createSingleMessage` is unused

(unused)